src/python/pants/engine/build_files.py

```diff
@@ -226,6 +244,7 @@ def addresses_from_address_families(address_families, spec):
   return Addresses(addresses)
 
 
+@rule(BuildDirs, [Select(AddressMapper), Select(Snapshot)])
 def filter_build_dirs(address_mapper, snapshot):
```

Comment:
I've thought about this a bit more, and I think I do like "rule" here, due at least in part to the similarity to prolog rules. And I think that people will think of the entire function and its annotation as the rule, so making the name more imperative/literal seems like to cause confusion.